core/src/main/java/org/mule/api/el/ExpressionLanguage.java

```diff
@@ -109,4 +110,5 @@
     @Deprecated
     <T> T evaluate(String expression, MuleMessage message, Map<String, Object> vars);
 
+    TypedValue evaluateTyped(String expression, MuleMessage message);
```

Comment:
The issue here is that this logic is needed on some transformers, and they work on the message, not the event.